app/lib/RouteResolver.js

```diff
@@ -2,16 +2,49 @@
 
 const fs = require('fs');
 const path = require('path');
+const parse = require('url').parse;
 const _ = require('lodash');
+const pathToRegExp = require('path-to-regexp');
 const expandPath = require('../../lib/expandPath');
 const Expectation = require('./Expectation');
+
+function isEqualMethod(method1, method2) {
+  const m1 = method1.toLowerCase();
+  const m2 = method2.toLowerCase();
+  return m1 === 'all' || m2 === 'all' || m1 === m2;
+}
+
+function isRouteForRequest(route, req) {
+  if (!isEqualMethod(req.method, route.method)) return false;
+
+  const pathname = parse(req.url, true).pathname;
+
+  if (route.pathname !== '*' && !route.pathRegExp.test(pathname)) return false;
+
+  // TODO: Later add features to match other things, like query parameters, etc.
+
+  return true;
+}
+
+function isRouteToReplace(newRoute, oldRoute) {
+  if (newRoute.pathname === oldRoute.pathname && newRoute.method === oldRoute.method) {
+    return true;
+  }
+
+  return false;
```

Comment:
You can return the result of the predicate here. I.e.:
```js
function isRouteToReplace(newRoute, oldRoute) {
  return newRoute.pathname === oldRoute.pathname && newRoute.method === oldRoute.method;
}
```

Reply:
@ryanricard Good point. I left it like this because this method will grow in future if we add other route matching features like query parameters, headers, request body, etc., as is the intent of the infrastructure in this refactoring. But I agree with you - for now, I've pushed up your simplification.